CinemaTicketSalesSystem/CinemaTicketSalesSystem/Views/Shared/MasterViewPartial.cshtml

```diff
@@ -1,60 +1,55 @@
-﻿@using Microsoft.AspNet.Identity
-@if (Request.IsAuthenticated)
-{
+﻿@using Microsoft.AspNet.Identity;
+@using CinemaTicketSalesSystem.Helpers;
+@using System.Security.Claims;
 
-    @Html.AntiForgeryToken()
-    if (User.IsInRole("Admin"))
-    {
-        <ul class="navbar-nav ml-auto">
-            <li class="nav-item">
-                @Html.ActionLink("Set Roles", "SetRoles", "Account")
-            </li>
-            <li class="nav-item">
-                @Html.ActionLink("Create New", "CreateNewMovie", "EditMovie")
-            </li>
-            <li class="nav-item">
-                @Html.ActionLink("Hello, " + User.Identity.GetUserName() + "!", "Index", "Home")
-            </li>
-            <li class="nav-item">
-                @Html.ActionLink("Logout", "Logout", "Account")
-            </li>
-        </ul>
-    }
-    else if (User.IsInRole("Moderator"))
-    {
-        <ul class="navbar-nav ml-auto">
-            <li class="nav-item">
-                @Html.ActionLink("Create New", "CreateNewMovie", "EditMovie")
-            </li>
-            <li class="nav-item">
-                @Html.ActionLink("Hello, " + User.Identity.GetUserName() + "!", "Index", "Home")
-            </li>
+
+@Html.AntiForgeryToken();
+@if (User.IsInRole("Admin"))
+{
+    var links = NavigationLinkHelper.GetNavigationLinks("Admin");
+    <ul class="navbar-nav ml-auto">
+        @foreach (var l in links)
+        {
             <li class="nav-item">
-                @Html.ActionLink("Logout", "Logout", "Account")
+                @Html.ActionLink(l.Label, l.Method, l.Controller)
             </li>
-        </ul>
-    }
-    else 
-    {
-        <ul class="navbar-nav ml-auto">            
+        }
+    </ul>
+}
+else if (User.IsInRole("Moderator"))
+{
+    var links = NavigationLinkHelper.GetNavigationLinks("Moderator");
+    <ul class="navbar-nav ml-auto">
+        @foreach (var l in links)
+        {
             <li class="nav-item">
-                @Html.ActionLink("Hello, " + User.Identity.GetUserName() + "!", "Index", "Home")
+                @Html.ActionLink(l.Label, l.Method, l.Controller)
             </li>
+        }
+    </ul>
+}
+else if (User.IsInRole("User"))
```

Comment:
you can use next statement
@{
List<  LinksViewModel  > links;
if (User.IsInRole(Constants.Admin)) {
    links = NavigationHelper.GetNavigationLinks(Constants.Admin);
.....
}
}

and then make a loop